articles/automation-testing-in-selenium/index.md

```diff
@@ -0,0 +1,99 @@
+Selenium is a software used for automation testing in new applications. This software provides a framework for the testing of web applications (across various web platforms). It is open-source, which means that developers will not incur costs related to licensing. 
+
+Automation testing in Selenium has helped in solving some of the challenges experienced in manual testing. In manual testing, the test cases are executed manually. The outcome of the test is compared with the expected or desired outcome. The disparity between the desired and actual outcome indicates the presence of a defect. This defect is fixed to ensure that the application is error-free. 
+
+Selenium conducts automation testing within a shorter time compared to manual testing. It also provides various tools that increase the scope of testing. In this type of automation testing, logs are created automatically, which increases the level of accuracy. 
+
+Some of the programming languages employed in creating text scripts in Selenium include Java, Python, C#, Perl, Ruby, and PhP. This software supports testing across various browsers such as Internet Explorer, Apple Safari, Mozilla Firefox, and Google Chrome. Automation testing in Selenium can also be done on various operating systems such as Linux, Windows, and Mac.
+
+### Why Selenium is Widely Used in Automation Testing
+Selenium is widely used by organizations and web developers in the testing of web applications. The following are some of the reasons for the adoption of this software in automation testing.
+* **Ease of use:** This software can be installed and used easily. 
+* **Low Cost:** This software is open-source, which reduces the cost of licensing and testing. This also provides an open community for providing customer support. 
+* **Flexibility:** Selenium software supports different browsers, programming languages, and browsers. This offers flexibility to web developers. This is unlike other testing software like QTP and RFT, which support fewer number of browsers.
+* **Integration:** It is possible to integrate this software with other applications such as TestNG, Docker, Maven, and Junit. These applications add value to automation testing, especially test management.
+* **Hardware consumption:** The execution of scripts requires low hardware consumption. 
+
+### Components of Selenium and How They Work
+Selenium software has four main components: Integrated Development Environment (IDE), remote control (RC), web driver, and grid. These components perform specific roles in automation testing.
+
+#### Selenium Integrated Development Environment (IDE)
+This can also be termed as a Selenium Recorder. It is utilized in the editing and recording of functional tests. It is also used for replaying and debugging tests. IDE has a playback feature that enables users to playback the recorded tests. In Mozilla Firefox, it is implemented as an add-on. In google chrome, it is made available as a plugin. 

```

Comment:
> This can also be termed as a Selenium Recorder. It is utilized in the editing and recording of functional tests.

What is a functional test? Please give a brief explanation.

Reply:
Done.